earth_enterprise/src/fusion/autoingest/sysman/InsetInfo.h

```diff
@@ -168,6 +168,60 @@ void CalcPacketGenInfo(const khTilespace            &tilespace,
                        const bool   is_overlay_terrain_proj,
                        const uint32 overlay_terrain_resources_min_level);
 
+// helper struct to make more generic calls from multiple locations possible
+// can be modified to allow for easier usage in possible new algorithms
+// for calculating overlap
+template <typename ProductAssetVersion>
+struct overlapEnvelope
+{
+    AssetDefs::Type& type;
+    khInsetCoverage gencov;
+    std::vector<const InsetInfo<ProductAssetVersion> *>& insets;
+    uint numInsets;
+    uint beginMinifyLevel;
+    uint endMinifyLevel;
+    uint level;
+    overlapEnvelope(AssetDefs::Type& _type,
+                    const khInsetCoverage& _gencov,
+                    std::vector<const InsetInfo<ProductAssetVersion> *>& _insets,
+                    uint _numInsets,
+                    uint _beginMinifyLevel, uint _endMinifyLevel, uint _level)
+
+   :                type(_type), gencov(_gencov), insets(_insets),
+                    numInsets(_numInsets),
+                    beginMinifyLevel(_beginMinifyLevel), endMinifyLevel(_endMinifyLevel),
+                    level(_level) {}
+};
+
+template <typename ProductAssetVersion>
+class OverlapCalculator
+{
+private:
+    std::vector<uint> neededIndexes;
+    void CalculateOverlap();
+
+public:
+    OverlapCalculator(overlapEnvelope<ProductAssetVersion> _env) :
+        env(_env) {}
+
+    overlapEnvelope<ProductAssetVersion>& env;
+
+    void PreprocessForInset()
+    {
+        neededIndexes.clear();
+        CalculateOverlap();
+    }
+
+    std::vector<uint> GetOverlapForLevel(bool recalculate = false)
```

Comment:
Minor: you could get rid of the boolean by just returning neededIndexes from PreprocessForInset and not calling GetOverlapForLevel at all from BuildIndex.